src/Microsoft.CodeAnalysis.Razor/CompilationTagHelperFeature.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class CompilationTagHelperFeature : ITagHelperFeature
+    {
+        private RazorEngine _engine;
+        private ITagHelperDescriptorProvider[] _providers;
+        private IMetadataReferenceFeature _referenceFeature;
+        
+        public RazorEngine Engine
+        {
+            get
+            {
+                return _engine;
+            }
+            set
+            {
+                _engine = value;
+                OnInitialized();
+            }
+        }
+
+        public IReadOnlyList<TagHelperDescriptor> GetDescriptors()
+        {
+            var results = new List<TagHelperDescriptor>();
+
+            var context = new Context(results);
+            var compilation = CSharpCompilation.Create("__TagHelpers", references: _referenceFeature.References);
```

Comment:
It might make sense to eventually have this as a feature. That way we can create a compilation once and re-use it across features if the need arises. No action now, just a shower thought 😄